packages/@aws-cdk/aws-glue/lib/view.ts

```diff
@@ -0,0 +1,152 @@
+import { ITable } from './table';
+
+/**
+ * View props.
+ *
+ * @experimental
+ */
+export interface ViewProps {
+  /**
+   * Name of the table.
+   */
+  readonly tableName: string;
+
+  /**
+   * Description of the table.
+   *
+   * @default generated
+   */
+  readonly description?: string;
+
+  /**
+   * Database in which to store the table.
+   */
+  readonly database: IDatabase;
+
+  /**
+   * Columns of the table.
+   */
+  readonly columns: Column[];
+
+  /**
+   * Statement to run for this view.
+   */
+  readonly statement: string;
+
+  /**
+   * Placeholders to replace in the statement.
+   *
+   * @default - `database` resolves to the databaseName of {database} and can't be overridden.
+   */
+  readonly placeHolders?: { [key: string]: string };
+}
+
+// the mapping of athena create table to PrestoDB types
+const prestoTypes: { [key: string]: string } = {
+  [Schema.FLOAT.inputString]: 'real',
+  [Schema.BIG_INT.inputString]: 'bigint',
+  [Schema.INTEGER.inputString]: 'integer',
+  [Schema.SMALL_INT.inputString]: 'smallint',
+  [Schema.TINY_INT.inputString]: 'tinyint',
+  [Schema.STRING.inputString]: 'varchar',
+  [Schema.BINARY.inputString]: 'varbinary',
+};
+
+/**
+ * A View in AWS Glue, implemented by an AWS Glue table.
+ *
+ * @experimental
+ * @resource AWS::glue::Table
+ */
+export class View extends Resource implements ITable {
+  /**
+   * Name of this table.
+   */
+  public readonly tableName: string;
+
+  /**
+   * ARN of this table.
+   */
+  public readonly tableArn: string;
+
+  /**
+   * @summary Creates a Table construct that represents a view in the AWS Glue data catalogue.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param props Properties.
+   */
+  constructor(scope: Construct, id: string, props: ViewProps) {
+    super(scope, id, {
+      physicalName: props.tableName,
+    });
+
+    const columns = props.columns;
+
+    const placeHolders = {};
+    if (props.placeHolders) {
+      Object.assign(placeHolders, props.placeHolders);
+    }
+    Object.assign(placeHolders, { database: props.database.databaseName });
+
+    const viewOriginalText = {
+      originalSql: props.statement,
+      catalog: 'awsdatacatalog',
+      columns: this.renderColumns(columns, true),
+      schema: '${database}',
+    };
+
+    const tableResource = new CfnTable(this, 'Table', {
+      catalogId: props.database.catalogId,
+      databaseName: props.database.databaseName,
+      tableInput: {
+        name: this.physicalName,
+        description: props.description || `${props.tableName} generated by CDK`,
+        parameters: { presto_view: true },
+        storageDescriptor: {
+          columns: this.renderColumns(columns),
+          serdeInfo: {},
+        },
+        partitionKeys: [],
+        tableType: 'VIRTUAL_VIEW',
+        viewOriginalText:
+          '/* Presto View: ' + Fn.base64(Fn.sub(JSON.stringify(viewOriginalText), placeHolders)) + ' */',
```

Comment:
I use placeholders since you will refer to other table names in your views from the template and can't put that into the statement injected via `Code.fromAsset`.

Reply:
Can you help me understand what the `Presto View` is and how we expect this line of code to work?

Reply:
Amazon Athena is built on Presto. Everything within `ViewOriginalText` is how Amazon Athena stores the view's SQL query and the datatypes. It's stored as created by a `CREATE VIEW` statement.